sources/packages/backend/libs/integrations/src/services/disbursement-schedule-errors/disbursement-schedule-errors.service.ts

```diff
@@ -61,12 +73,40 @@ export class DisbursementScheduleErrorsService extends RecordDataModelService<Di
       disbursementFeedbackError.creator = auditUser;
       return disbursementFeedbackError;
     });
-    return this.repo
-      .createQueryBuilder()
-      .insert()
-      .into(DisbursementFeedbackErrors)
-      .values(disbursementFeedbackErrors)
-      .orIgnore()
-      .execute();
+    const user =
+      disbursementSchedule.studentAssessment.application.student.user;
+    const ministryNotification: ECertFeedbackFileErrorNotification = {
+      givenNames: user.firstName,
+      lastName: user.lastName,
+      applicationNumber:
+        disbursementSchedule.studentAssessment.application.applicationNumber,
+      errorCodes: blockFundingErrorIds,
+    };
+    const metadata = {
+      disbursementId: disbursementSchedule.id,
+      documentNumber: disbursementSchedule.documentNumber,
+      applicationNumber:
+        disbursementSchedule.studentAssessment.application.applicationNumber,
+      errorCodes: blockFundingErrorIds,
+    };
+    return this.dataSource.transaction(
+      async (transactionalEntityManager: EntityManager) => {
+        if (blockFundingErrorIds.length) {
+          await this.notificationActionsService.saveEcertFeedbackFileErrorNotification(
+            ministryNotification,
```

Comment:
`ministryNotification` can be created inside the `if(blockFundingErrorIds.length)` because it will not be used otherwise.